### PR TITLE
add test file for autoscaling defaults

### PR DESCRIPTION
**Pull request by yarntime**
add test file `pkg/apis/autoscaling/v1/defaults_test.go`

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34414)

### CODE REVIEW

**Comment by yarntime** [comment]
add cla/linuxfoundation

File: pkg/apis/autoscaling/v1/defaults_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			test:           "set min replicas to 3",
+		},
+	}
+
+	for _, test := range tests {
+		hpa := &test.hpa
+		obj2 := roundTrip(t, runtime.Object(hpa))
+		hpa2, ok := obj2.(*HorizontalPodAutoscaler)
+		if !ok {
+			t.Errorf("unexpected object: %v", hpa2)
```
Comment (jszczepkowski) [inline pkg/apis/autoscaling/v1/defaults_test.go]: nit: use t.Fatalf

Comment (jszczepkowski) [inline pkg/apis/autoscaling/v1/defaults_test.go]: I would rather print here obj2 than hpa2, which may be nil

File: pkg/apis/autoscaling/v1/defaults_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		},
+	}
+
+	for _, test := range tests {
+		hpa := &test.hpa
+		obj2 := roundTrip(t, runtime.Object(hpa))
+		hpa2, ok := obj2.(*HorizontalPodAutoscaler)
+		if !ok {
+			t.Fatalf("unexpected object: %v", obj2)
+			t.FailNow()
```
Comment (jszczepkowski) [inline pkg/apis/autoscaling/v1/defaults_test.go]: FailNow is no longer needed, as Fatalf is used.

**Comment by jszczepkowski** [comment]
LGTM

**Comment by yarntime** [comment]
Hi @jszczepkowski ,  can you give the "ok to test" comment? thank you very much.

**Comment by jszczepkowski** [comment]
ok to test

**Comment by jszczepkowski** [comment]
@k8s-bot ok to test

**Comment by yarntime** [comment]
@k8s-bot cvm gke e2e test this

**Comment by jszczepkowski** [comment]
It seems that golint is failing:
```
hack/make-rules/../../hack/verify-golint.sh
```
@yarntime Can you fix it?

**Comment by yarntime** [comment]
@k8s-bot verify test this

**Comment by yarntime** [comment]
@jszczepkowski , could you give the LGTM label? thanks.

**Comment by jszczepkowski** [comment]
LGTM

**Comment by yarntime** [comment]
Hi @jszczepkowski , @lavalamp , could you help me to merge this pr? thank you very much.

**Comment by jszczepkowski** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**